package edu.towson.cis.cosc442.project4.coffeemaker;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import org.junit.Test;

import junit.framework.TestCase;

/**
 *
 */
public class CoffeeMakerTest extends TestCase {
	private CoffeeMaker cm;
	private Inventory i;
	private Recipe r1;

	public void setUp() {
		cm = new CoffeeMaker();
		i = cm.checkInventory();

		r1 = new Recipe();
		r1.setName("Coffee");
		r1.setPrice(50);
		r1.setAmtCoffee(6);
		r1.setAmtMilk(1);
		r1.setAmtSugar(1);
		r1.setAmtChocolate(0);
	}

	public void testAddRecipe1() {
		assertTrue(cm.addRecipe(r1));
	}

	public void testDeleteRecipe1() {
		cm.addRecipe(r1);
		assertTrue(cm.deleteRecipe(r1));
	}

	public void testEditRecipe1() {
		cm.addRecipe(r1);
		Recipe newRecipe = new Recipe();
		newRecipe = r1;
		newRecipe.setAmtSugar(2);
		assertTrue(cm.editRecipe(r1, newRecipe));
	}
	
	public void testAddInventory1() {
		assertTrue(cm.addInventory(30, 10, 10, 10));
	}
	
	public void testAddInventory2() {
		assertFalse(cm.addInventory(30, -10, 10, 10));
	}
	
	public void testAddInventory3() {
		assertFalse(cm.addInventory(-30, 10, 10, 10));
	}
	
	public void testAddInventory4() {
		assertFalse(cm.addInventory(30, 10, -10, 10));
	}
	
	public void testAddInventory5() {
		assertFalse(cm.addInventory(30, 10, 10, -10));
	}
	
	public void testAddInventory6() {
		Inventory in = new Inventory();
		int i = in.getCoffee();
		
		cm.addInventory(1,1,1,1);
		int j = in.getCoffee();
		
		if(j>i){
			fail();
		} 
	}
	
	public void testAddInventory7() {
		Inventory in = new Inventory();
		int i = in.getChocolate();
		
		cm.addInventory(1,1,1,1);
		int j = in.getChocolate();
		
		if(j>i){
			fail();
		} 
	}
	
	public void testCheckInventory1() {
		try{
			cm.checkInventory();
		} catch(Exception e){}
	}
	public void testpurchaseBeverage1() {
		cm.addRecipe(r1);
		
		int amtPaid = 50;
		assertEquals(cm.makeCoffee(r1, amtPaid), 0, .001);
	}
	
	/**
	 * Run the Recipe getRecipeForName(String) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 3/29/16 11:09 AM
	 */
	@Test
	public void testGetRecipeForName_1()
		throws Exception {
		CoffeeMaker fixture = new CoffeeMaker();
		String name = "";

		Recipe result = fixture.getRecipeForName(name);

		// add additional test code here
		assertNotNull(result);
		assertEquals(null, result.toString());
		assertEquals(null, result.getName());
		assertEquals(0, result.getAmtMilk());
		assertEquals(0, result.getAmtSugar());
		assertEquals(0, result.getAmtChocolate());
		assertEquals(0, result.getAmtCoffee());
		assertEquals(0, result.getPrice());
	}
}